cmd/start.go

```diff
@@ -96,7 +105,26 @@ var startCmd = &cobra.Command{
 			<-c
 			logger.Info("stopped message processor")
 			return nil
-		})
+		}
+
+		if embedNats {
+			opts := &server.Options{}
+			ns, err := server.NewServer(opts)
+
+			if err != nil {
+				panic(err)
+			}
+
+			go ns.Start()
+
+			for !ns.ReadyForConnections(4 * time.Second) {
+				logger.Info("Waiting for nats server to start...")
+			}
+			logger.Info("Nats server started at url: %s", ns.ClientURL())
+			runProvider(logger, ns.ClientURL(), dryRun, runProviderCallback)
+		}
```

Comment:
put the other run in an else?

Reply:
We resolved this by using a waitgroup and running the blocking provider in a go routine.